src/main/groovy/io/seqera/wave/service/metric/impl/MetricsServiceImpl.groovy

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Wave, containers provisioning service
+ *  Copyright (c) 2024, Seqera Labs
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.seqera.wave.service.metric.impl
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import io.seqera.wave.service.metric.MetricConstants
+import io.seqera.wave.service.metric.MetricsCounterStore
+import io.seqera.wave.service.metric.MetricsService
+import io.seqera.wave.tower.PlatformId
+import jakarta.inject.Inject
+import jakarta.inject.Singleton
+/**
+ * Implements service to store and retrieve wave metrics from the counter store
+ *
+ * @author Munish Chouhan <munish.chouhan@seqera.io>
+ */
+@Slf4j
+@Singleton
+@CompileStatic
+class MetricsServiceImpl implements MetricsService {
+
+    @Inject
+    private MetricsCounterStore metricsCounterStore
+
+    DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+
+    @Override
+    Long getBuildsMetrics(String date, String org) {
+        return metricsCounterStore.get(getKey(MetricConstants.PREFIX_BUILDS, date, org)) ?: 0
+    }
+
+    @Override
+    Long getPullsMetrics(String date, String org) {
+        return metricsCounterStore.get(getKey(MetricConstants.PREFIX_PULLS, date, org)) ?: 0
+    }
+
+    @Override
+    Long getFusionPullsMetrics(String date, String org) {
+        return metricsCounterStore.get(getKey(MetricConstants.PREFIX_FUSION, date, org)) ?: 0
+    }
+
+    @Override
+    void incrementFusionPullsCounter(PlatformId platformId){
+        incrementCounter(MetricConstants.PREFIX_FUSION, platformId?.user?.email)
+    }
+
+    @Override
+    void incrementBuildsCounter(PlatformId platformId){
+        incrementCounter(MetricConstants.PREFIX_BUILDS, platformId?.user?.email)
+    }
+
+    @Override
+    void incrementPullsCounter(PlatformId platformId) {
+        incrementCounter(MetricConstants.PREFIX_PULLS, platformId?.user?.email)
+    }
+
+    protected void incrementCounter(String prefix, String email) {
+        def org = getOrg(email)
```

Comment:
It looks like the `org` can be `null`, but in that case the `if` branch is skipped at all. It that correct?

Reply:
if org is null, so only the date key will be incremented and if its not null, org and org+date key will be incremented

Reply:
ok, now I see it